lib/active_interaction/base.rb

```diff
@@ -234,6 +233,22 @@ def run_validations!
 
     private
 
+    # We want to allow both `Hash` objects and `ActionController::Parameters`
+    # objects. In Rails < 5, parameters are a subclass of hash and calling
+    # `#symbolize_keys` returns the entire hash, not just permitted values. In
+    # Rails >= 5, parameters are not a subclass of hash but calling
+    # `#to_unsafe_h` returns the entire hash.
+    def normalize_inputs!(inputs)
+      return inputs if inputs.is_a?(Hash)
+
+      klass = 'ActionController::Parameters'.safe_constantize
+      if !klass || !inputs.is_a?(klass)
+        raise ArgumentError, 'inputs must be a hash'
+      end
```

Comment:
Right now the code goes, good, bad, good. It might work better to group the good paths.

```ruby
return inputs if inputs.is_a?(Hash)

klass = 'ActionController::Parameters'.safe_constantize
if klass && inputs.is_a?(klass)
  return inputs.to_unsafe_h
end

raise ArgumentError, 'inputs must be a hash'
```